### PR TITLE
[python] Add resource plugin for python, dataX, CustomDataX and Sql

### PR DESCRIPTION
**Pull request by xdu-chenrj**
## Purpose of the pull request

Add resource plugin support to task python, dataX, CustomDataX and Sql #10911 

mail-list
https://lists.apache.org/thread/r1f9wwlrj6jrnyc1hwpf8qoq8zr8n7j1

## Brief change log

Add resource plugin support to task python, dataX, CustomDataX and Sql

Add resource plugin test of task task python, dataX, CustomDataX and Sql

### CODE REVIEW

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py]: we should not remove the statement `self.definition = definition`, I think we should better use 

```python
    # For extension record
    ext: set = {".py"}
    ext_attr: str = "_definition"

    def __init__(
         self, name: str, definition: Union[str, types.FunctionType], *args, **kwargs
     ):
         self._raw_script = self.raw_script(definition)
         super().__init__(name, TaskType.PYTHON, *args, **kwargs)
         self._definition = definition
```

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py]: And keep  other setting in the old style

**Comment by zhongjiajie** [comment]
base on #12025, do not merge until #12025 merged

**Comment by zhongjiajie** [comment]
@xdu-chenrj  can you rebase on upstream/dev to solve the conflict? it seems some content is already in other PR

**Comment by zhongjiajie** [comment]
can you rebase on upstream/dev to solve the conflict? it seems some content is already in other PR

**Comment by xdu-chenrj** [comment]
> @xdu-chenrj can you rebase on upstream/dev to solve the conflict? it seems some content is already in other PR

ok

**Comment by xdu-chenrj** [comment]
> @xdu-chenrj can you rebase on upstream/dev to solve the conflict? it seems some content is already in other PR

It seems that there is no conflict.

File: dolphinscheduler-python/pydolphinscheduler/tests/tasks/test_sql.py
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/tests/tasks/test_sql.py]: nit, but can we change content to SQL query instead of shell?

File: dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py
Comment (zhongjiajie) [inline dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py:78]: we can directly use `self.definition` here

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py:78]: It is same as `definition = getattr(self, "definition")`

Comment (zhongjiajie) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py:78]: we do you change those, does it raise error?

Comment (xdu-chenrj) [reply dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py:78]: If the `self.definition` is used directly, a warning of `unresolved attribute` will be reported, although it can also run normally.

In init fun, we change the `definition` attribute to `_definition`

Through the `getattr` method, the `definition` attribute will not have a warning.

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**
